### PR TITLE
Help button

### PR DESCRIPTION
**Pull request by maxwang7**
Issue: codalab/codalab-worksheets#251
Corresponding PR in codalab-worksheets: https://github.com/codalab/codalab-cli/pull/675#pullrequestreview-20198755

Creates a user workflow for sending support emails to CodaLab's support email address. You can view the UI at: http://138.197.208.207:8000/. (My deployment)

Some screenshots:

Initial state:

screen shot 2017-02-05 at 9 51 32 pm

After clicking on the help button:

screen shot 2017-02-05 at 9 52 46 pm

### CODE REVIEW

**Comment by maxwang7** [comment]
Also, question: Should I put all of this functionality in a flag?

**Comment by kashizui** [comment]
@maxwang7 Do you have emailing set up on your server? I didn't get a verification email after signing up.

And what do you mean by "in a flag"?

**Comment by maxwang7** [comment]
@kashizui 
Re email: No I don't, so my emails are printing to the console. Here's your email:

========================================
From: console
To: sckoo@cs.stanford.edu
Subject: Verify your CodaLab account

Hi sckoo,

You have signed up for an CodaLab account on 138.197.208.207:8000 using this
email address.  To confirm and activate your account, please follow the link
below:

http://138.197.208.207:8000/rest/account/verify/30817e31fbf043e5951d8e80a97aa266

Enjoy!

The CodaLab Team

========================================

Re flag: I meant should we make it so that there's a configuration variable that will turn the feature on or off? I ask because Eric hid his chat feature using a configuration variable/flag.

**Comment by kashizui** [comment]
Thanks!

Oh I see, my personal opinion is that a flag isn't necessary for this since it won't require lots of additional development before we can just deploy it, @percyliang thoughts?

**Comment by kashizui** [comment]
@maxwang7 Hm I don't see the help button?
screen shot 2017-02-05 at 10 39 27 pm

**Comment by maxwang7** [comment]
Oh sorry, I switched to another branch, working on debugging the keyboard navigation. Give me a second, will switch it back.

**Comment by maxwang7** [comment]
Ok, it's good to go!

**Comment by percyliang** [comment]
Don't think we need a flag either.

File: codalab/apps/web/static/js/worksheet/worksheet_help_button.jsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+        var onSuccess = function(data, status, jqXHR) {
+            console.log("success!");
+            console.log(data);
+            this.messageSentTransition(true);
+        }.bind(this);
+
+        var onError = function(jqHXR, status, error) {
+            console.log("error");
+            console.log(error);
```
Comment (kashizui) [inline codalab/apps/web/static/js/worksheet/worksheet_help_button.jsx]: I think our convention for error handling is just:
```
console.error(jqXHR.responseText);
// OR if it needs to be visible, something like
$("#error-message").html(xhr.responseText);
// or something more React-ish
```
and the backend should ensure that the body of the response in the case of a error should be a simple plaintext error message. This is already automatically the case in most situations when an exception gets thrown on the server side.

**Comment by maxwang7** [comment]
Re style: I was talking with a friend today who's a really experienced web designer. She recommended making all of the buttons rounded. I was intending to bring this up next meeting (along with some other UI stuff), so I can leave the rounded corners and hopefully we'll have another commit that will change the rest of the buttons to rounded corners?

Re button bevels: Yes, let me look into that, I'm not sure why that's happening.

**Comment by kashizui** [comment]
Yea I don't have a super strong opinion, just that things are consistent.

**Comment by maxwang7** [comment]
The button has been updated (there's also a shadow now!), here's a screenshot:

screen shot 2017-02-07 at 4 17 08 pm

File: codalab/apps/web/static/js/worksheet/worksheet_help_button.jsx (hunk 1 above)
Comment (maxwang7) [inline codalab/apps/web/static/js/worksheet/worksheet_help_button.jsx]: Ok, let me change that, thanks!

**Comment by kashizui** [comment]
shadow looks pretty cool!

**Comment by kashizui** [comment]
You can go ahead and merge when you think it's ready

**Pull request merged**